generic_permissions/visibilities.py

```diff
@@ -44,14 +51,83 @@ def get_queryset(self):
         return queryset
 
 
+class VisibilityManyRelatedField(ManyRelatedField):
+    def get_attribute(self, instance):
+        queryset = super().get_attribute(instance)
+        for handler in VisibilitiesConfig.get_handlers(queryset.model):
+            queryset = handler(queryset, self.parent._context["request"])
+
+        return queryset
+
+
+class VisibilityRelatedFieldMixin:
+    @classmethod
+    def many_init(cls, *args, **kwargs):
+        # ManyToManyField
+        list_kwargs = {"child_relation": cls(*args, **kwargs)}
+        for key in kwargs:
+            if key in MANY_RELATION_KWARGS:
+                list_kwargs[key] = kwargs[key]
+        return VisibilityManyRelatedField(**list_kwargs)
+
+    def get_queryset(self):  # pragma: no cover
+        # TODO not sure when get_queryset is called, but it is needed
```

Comment:
:thinking:

Reply:
Leftover from testing, this method isnt needed for the visibility checking. Removed it